File: app/src/main/java/com/drivingschoolrwandaapp/database/entities/TestWithQuestions.java
package com.drivingschoolrwandaapp.database.entities;

import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;

public class TestWithQuestions {
    @Embedded
    public TestEntity test;

    @Relation(
        entity = TestQuestionEntity.class,
        parentColumn = "id",
        entityColumn = "testId"
    )
    public List<QuestionWithOptions> questions;
}

File: app/src/main/java/com/drivingschoolrwandaapp/database/entities/TestWithQuestions.kt
package com.drivingschoolrwandaapp.database.entities

import androidx.room.Embedded
import androidx.room.Relation

data class TestWithQuestions(
    @JvmField @Embedded var test: TestEntity? = null,
    @JvmField @Relation(
        entity = TestQuestionEntity::class,
        parentColumn = "id",
        entityColumn = "testId"
    )
    var questions: List<QuestionWithOptions>? = null
)
